### ISSUE TITLE
Use HttpPost for Logout

### ISSUE THREAD

**Issue by SpecialHias** [IdentityServer/IdentityServer3#2346]
Is there any way to logout of the IdentityServer3 with a HttpPost method?

I know that in the latest version of the IdentityServer3 the endpoint supports HttpPost, but I'm not able to do so from my MVC5 application.

Currently I'm using this method to log out:
public ActionResult Logout()
{
    Request.GetOwinContext().Authentication.SignOut();
    return Redirect("/");
}

Which will execute a HttpGet action.

**Comment by leastprivilege** [comment]
The middleware does not support it - you would need to do your own POST.

**Comment by SpecialHias** [comment]
Ok. I thought so ...
Thanks for your quick response!

If i execute a post action to: https://baseuri/connect/endsession
with the following parameters:
* id_token_hint
* post_logout_redirect_uri

I will receive a successful logout message and will be redirected back to my application, but the user is not signed out.
Do i need another parameter?

**Comment by leastprivilege** [comment]
that is weird.

check the logs.

**Comment by SpecialHias** [comment]
The logs for the HttpGet and HttpPost logout look pretty equal.

I've found no error in HttpPost version (a truncated snapshot from the logs):
[IdentityServer3.Core.Configuration.Hosting.CorsPolicyProvider] CORS request made for path: /connect/endsession from origin: https://localhost:44302 but rejected because invalid CORS path
[IdentityServer3.Core.Endpoints.EndSessionController] Start end session request
[IdentityServer3.Core.Validation.EndSessionRequestValidator] Start end session request validation
[IdentityServer3.Core.Validation.TokenValidator] Start identity token validation
[IdentityServer3.Core.Validation.TokenValidator] Token validation success
[IdentityServer3.Core.Validation.EndSessionRequestValidator] End session request validation success
[IdentityServer3.Core.Endpoints.EndSessionController] End end session request
[IdentityServer3.Core.Results.LogoutResult] Redirecting to logout page
[IdentityServer3.Core.Configuration.Hosting.MessageCookie`1] Protecting message: {"ClientId":"MyClientId","ReturnUrl":"https://localhost:44302/Home","Created":635869774953799417}
[IdentityServer3.Core.Endpoints.AuthenticationController] Logout endpoint submitted
[IdentityServer3.Core.Endpoints.AuthenticationController] Clearing cookies
[IdentityServer3.Core.Endpoints.AuthenticationController] rendering logged out page
[IdentityServer3.Core.Endpoints.EndSessionController] End session callback requested
[IdentityServer3.Core.Endpoints.EndSessionController] No client end session iframe URLs

**Comment by SpecialHias** [comment]
If somebody is struggling with the same problem, here is my solution:
I wrote my own form which posts all necessary information to the endsession endpoint and afterwards i had to destroy the authentication cookie. 

IdentityServer3 works as expected but the OWIN middleware is not ready for HttpPosts.

I also described my [solution on stackoverflow](http://stackoverflow.com/questions/34349774/use-httppost-for-logout-in-owin-katana-authentication-manager).
